jujugui/static/gui/src/app/components/entity-details/content/diagram/diagram.js

```diff
@@ -19,29 +19,67 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 'use strict';
 
 class EntityContentDiagram extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      url: this.props.getDiagramURL(this.props.id)
```

Comment:
Because this is never used more than once on load I don't see why we couldn't require that a valid url be passed into the component. Dropping the requirement for `state` and for the `getDiagramURL` prop.